src/dawn/IIR/MetaInformation.h

```diff
@@ -0,0 +1,493 @@
+
+  /// Injection of AccessIDs of literal constant to their respective name (usually the name is just
+  /// the string representation of the value). Note that literals always have *strictly* negative
+  /// AccessIDs, which makes them distinguishable from field or variable AccessIDs. Further keep in
+  /// mind that each access to a literal creates a new AccessID!
+  std::unordered_map<int, std::string> LiteralAccessIDToNameMap_;
+
+  /// This is a set of AccessIDs which correspond to fields. This allows to fully identify if a
+  /// AccessID is a field, variable or literal as literals have always strictly negative IDs and
+  /// variables are neither field nor literals.
+  std::set<int> FieldAccessIDSet_;
+
+  /// This is an ordered list of IDs of fields that belong to the user API call of the program
+  std::vector<int> apiFieldIDs_;
+
+  /// Set containing the AccessIDs of fields which are represented by a temporary storages
+  std::set<int> TemporaryFieldAccessIDSet_;
+
+  /// Set containing the AccessIDs of fields which are manually allocated by the stencil and serve
+  /// as temporaries spanning over multiple stencils
+  std::set<int> AllocatedFieldAccessIDSet_;
+
+  /// Set containing the AccessIDs of "global variable" accesses. Global variable accesses are
+  /// represented by global_accessor or if we know the value at compile time we do a constant
+  /// folding of the variable
+  std::set<int> GlobalVariableAccessIDSet_;
+
+  /// Map of AccessIDs to the list of all AccessIDs of the multi-versioned field, for fields and
+  /// variables
+  VariableVersions variableVersions_;
+
+  /// Stencil description statements. These are built from the StencilDescAst of the sir::Stencil
+  std::vector<std::shared_ptr<Statement>> stencilDescStatements_;
+  std::unordered_map<std::shared_ptr<StencilCallDeclStmt>, int> StencilCallToStencilIDMap_;
+  std::unordered_map<int, std::shared_ptr<StencilCallDeclStmt>> IDToStencilCallMap_;
+
+  /// StageID to name Map. Filled by the `PassSetStageName`.
+  std::unordered_map<int, std::string> StageIDToNameMap_;
+
+  /// Referenced stencil functions in this stencil (note that nested stencil functions are not
+  /// stored here but rather in the respecticve `StencilFunctionInstantiation`)
+  std::vector<std::shared_ptr<StencilFunctionInstantiation>> stencilFunctionInstantiations_;
+  std::unordered_map<std::shared_ptr<StencilFunCallExpr>,
+                     std::shared_ptr<StencilFunctionInstantiation>>
+      ExprToStencilFunctionInstantiationMap_;
+
+  /// lookup table containing all the stencil function candidates, whose arguments are not yet bound
+  std::unordered_map<std::shared_ptr<StencilFunctionInstantiation>,
+                     StencilFunctionInstantiationCandidate>
+      stencilFunInstantiationCandidate_;
+
+  /// BoundaryConditionCall to Extent Map. Filled my `PassSetBoundaryCondition`
+  std::unordered_map<std::shared_ptr<BoundaryConditionDeclStmt>, Extents>
```

Comment:
I think that iir::Stencil should contain a iir::BoundaryCondition, and that one contain this map

Reply:
why should this be on the level of the iir::Stencil? I think this should be on the level of each user-stencil has this associated with it. and this is the level of the steniclInstantiation